server/app.py

```diff
@@ -0,0 +1,129 @@
+# app.py
+from flask import Flask, request, jsonify
+from werkzeug.utils import secure_filename
+import os
+from PIL import Image
+import logging
+from flask_cors import CORS
+
+app = Flask(__name__)
+
+# Simplified CORS setup
+CORS(app, origins=['http://localhost:5173'])
+
+# Configure logging
+logging.basicConfig(level=logging.INFO)
+logger = logging.getLogger(__name__)
+
+# Configuration
+UPLOAD_FOLDER = 'uploads'
+ALLOWED_EXTENSIONS = {'png', 'jpg', 'jpeg'}
+MAX_CONTENT_LENGTH = 16 * 1024 * 1024
+
+app.config['UPLOAD_FOLDER'] = UPLOAD_FOLDER
+
+@app.route('/', methods=['GET'])
+def home():
+    return jsonify({
+        'status': 'success',
+        'message': 'Server is running'
+    })
+
+def allowed_file(filename):
+    return '.' in filename and filename.rsplit('.', 1)[1].lower() in ALLOWED_EXTENSIONS
+
+def get_file_size(file_path):
+    return os.path.getsize(file_path)
+
+def process_image(file_path):
+    try:
+        with Image.open(file_path) as img:
+            mock_upscale_factor = 2.0
+            return {
+                'success': True,
+                'mock_upscale_factor': mock_upscale_factor
+            }
+    except Exception as e:
+        logger.error(f"Error processing image: {str(e)}")
+        return {
+            'success': False,
+            'error': str(e)
+        }
+
+@app.route('/upload', methods=['POST', 'OPTIONS'])
+def upload_file():
```

Comment:
looks like we can split this function into two by handling OPTIONS and POST separately. That will reduce number of nested if statements.